internal/io/fs/move.go

```diff
@@ -0,0 +1,28 @@
+package fs
+
+import (
+	"fmt"
+	"os"
+)
+
+// SimpleAtomicMove performs an atomic move operation from src to dst.
+// If the destination already exists, it removes it before attempting the move.
+// It returns an error if the move operation fails for any reason, including
+// if the source file does not exist or if the destination cannot be created.
+func SimpleAtomicMove(src, dst string) error {
```

Comment:
Minimal copy because https://github.com/akuity/kargo/pull/4662 does the same (but includes tests, etc.)